### PR TITLE
feat(Table): normalize string for search in tables

### PR DESCRIPTION
**Pull request by matheushahnn**
Feature requested https://github.com/buefy/buefy/issues/3578

## Proposed Changes

- Normalize string for search in tables

Example using filter in table to find the last name with accent
image

### CODE REVIEW

File: src/utils/helpers.js [hunk 1]
```diff
@@ -174,6 +174,15 @@ export function escapeRegExpChars(value) {
...
     return value.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&')
 }
+/**
+ * Remove accents/diacritics in a string in JavaScript
+ * https://stackoverflow.com/a/37511463
+ */
+export function removeDiacriticsFromString(value) {
+    if (!value) return value
+
+    return value.normalize('NFD').replace(/[\u0300-\u036f]/g, '')
```
Comment (matheushahnn) [inline src/utils/helpers.js:184]: I'd say the best approach would be using `value.normalize("NFD").replace(/\p{Diacritic}/gu, "")` but looks like babel has problems with this regex https://github.com/babel/babel/issues/8951.

File: src/components/table/Table.vue [hunk 2]
```diff
@@ -1169,7 +1169,8 @@ export default {
                         if (value !== Number(input)) return false
                     } else {
                         const re = new RegExp(escapeRegExpChars(input), 'i')
-                        if (!re.test(value)) return false
+                        const valueWithoutDiacritics = removeDiacriticsFromString(value)
+                        if (!re.test(valueWithoutDiacritics) && !re.test(value)) return false
```
Comment (matheushahnn) [inline src/components/table/Table.vue:1173]: It keeps filtering if I search using accents or not. 
I thought to create a new param in the column to enable the filter with accents, but not sure if it is necessary. 
What do you think about it?

Comment (geeksilva97) [reply src/components/table/Table.vue:1173]: `OR` could be enough here?

```suggestion
        return !re.test(valueWithoutDiacritics) || !re.test(value)
```

Comment (matheushahnn) [reply src/components/table/Table.vue]: Nice catch!
I just need to remove the `!` in the condition and it worked.

Comment (antoniopetlove) [reply src/components/table/Table.vue]: Cool!

**Comment by jtommy** [comment]
@matheushahnn thanks 👍

**Comment by MariusRedseed** [comment]
Hey team, this breaks searching in tables when the value is an array of strings.

**Comment by jtommy** [comment]
@MariusRedseed array?

**Comment by MariusRedseed** [comment]
@jtommy yup. This assumes the value in the column is a string. The pull request below attempted to allow the value of a column to be an array, but still be searchable:
https://github.com/buefy/buefy/pull/2723

**Comment by jtommy** [comment]
Mmm you are right but it's no a good solution.. we need a field prop to filter the correct text

**Comment by MariusRedseed** [comment]
Sure thing, I used a custom search for now, for a column containing an array of strings:
```
:custom-search="
    ({column_with_array_values}, input) => {
        const re = new RegExp(input, 'i')
        return column_with_array_values.some(role => re.test(role));
    }"
```

**Comment by jtommy** [comment]
@MariusRedseed just restored

**Comment by AN-WM** [comment]
This addition breaks the ability to search numbers with decimals. I fixed it for myself by adjusting the removeDiacriticsFromString() function to be:
`export function removeDiacriticsFromString(value) {
    if (!value) return value

    //Only normalize when value type === string
    if (typeof value === 'string' || value instanceof String) {
        return value.normalize('NFD').replace(/[\u0300-\u036f]/g, '');
    } else {
        return value;
    }
}`

Screenshot buefy search decimal
Screenshot error buefy search decimal

**Pull request merged**